game/game.go

```diff
@@ -711,16 +715,47 @@ func (g *Game) playActivatedAbility() {
 }
 
 // Returns a binary representation of the game, json for now
-func (g *Game) serialized() []byte {
+func (g *Game) Serialized() []byte {
 	gameJson, _ := json.Marshal(g)
 	return gameJson
 }
 
 // Reset the game pointer for game.players, and return a deserialized game
-func deserializeGameState(jsonBytes []byte) *Game {
+func DeserializeGameState(jsonBytes []byte) *Game {
 	game := &Game{}
 	json.Unmarshal(jsonBytes, game)
 	game.Players[0].game = game
 	game.Players[1].game = game
+	for _, perm := range game.Permanents {
+		perm.game = game
+	}
 	return game
 }
+
+func CopyGame(g *Game) *Game {
+	newGame := &Game{}
+	copier.Copy(&newGame, &g)
```

Comment:
This doesn't seem like it will work, the permanents' pointers don't get reset for one. It's also not clear to me which pointers this "copier" library will use. I suggest either using serialization/deserialization or writing your own "copy" methods on things.